File: backend/app/schemas/ai.py
from typing import List, Optional
from pydantic import BaseModel, Field

class ChatRequest(BaseModel):
    message: str = Field(
        ...,
        min_length=1,
        max_length=2000,
        description="Farmer's agricultural query (max 2000 characters)"
    )

class ChatResponse(BaseModel):
    success: bool
    answer: str
    language: str
    sources: List[str] = []
    message: Optional[str] = None

File: frontend/src/types/ai.ts
export interface ChatMessage {
  id: string;
  sender: 'user' | 'assistant';
  text: string;
  language?: string;
  timestamp: string;
  sources?: string[];
  isError?: boolean;
}

export interface ChatRequest {
  message: string;
}

export interface ChatResponse {
  success: boolean;
  answer: string;
  language: string;
  sources: string[];
  message?: string | null;
}
